File endpoints

adupuis2 opened the pull request:
Added endpoints for uploading files and viewing a user's files.

There is currently a bug where uploading a file with a given file name multiple times will create multiple records for the one file on disk. I will probably add an index to `Files.name` and implement some logic to prevent duplicate records being created.

This builds atop the User endpoints, so will be marked as a draft PR until the dependencies are merged.

adupuis2 committed:
Create user endpoints and user sessions

adupuis2 committed:
Create user endpoints and user sessions

adupuis2 committed:
Create user endpoints and user sessions

adupuis2 committed:
Create file endpoints
Add support for file uploads

adupuis2 commented on src/rickbox/resources/file_manager.py:
```diff
@@ -0,0 +1,66 @@
+    @conceal_error_message
+    @require_session
+    def delete(self):
+        pass
```
Wasn't sure if we wanted to actually delete the files from within the Flask code, or look for/build a small external tool that might be safer.

jasonncleveland replied:
Python has robust libraries to interact with the file system. I think we're fine deleting files from within the code.

adupuis2 commented:
I might have forgotten to add authenticated/authorized file retrieval.
Implementing that now.

jasonncleveland commented on src/rickbox/app_config.py:
```diff
@@ -4,4 +4,24 @@
 MYSQL_PASS = 'password'
 MYSQL_DB = 'rickbox'
 
+APP_HOST = 'info3103.cs.unb.ca'
```
Duplicate keys

jasonncleveland commented on src/rickbox/resources/file_manager.py:
```diff
@@ -0,0 +1,66 @@
+        if file.filename == '':
+            return ERROR_400
+
+        # prepend filenames with user IDs to avoid naming conflicts
```
What if we stored each user's files in their own folder with the folder named as the user id. This would avoid having to prepend the filename.

adupuis2 committed:
Create endpoints for downloading, deleting, and renaming files
Create stored procedure for updating file records
Updated docstrings
Added new comments to some small functions
Added user-specific directories
Abstracted more responses
Rename UserSession -> SessionManager

adupuis2 committed:
Removed 'hello world' root route

adupuis2 commented:
These commits / timestamps...

![image](https://user-images.githubusercontent.com/19765648/76672204-b4622300-6579-11ea-9912-d734690633eb.png)


```
commit 9632da354f372c76a3919083d792bd99e86ae293
Author: Alain Dupuis <adupuis2@unb.ca>
Date:   Sun Mar 8 17:51:49 2020 -0300

    Create user endpoints and user sessions

commit 8a570f6157049a9d4137e6d6600532d0d4fe29da
Author: Alain Dupuis <adupuis2@unb.ca>
Date:   Sun Mar 8 17:51:49 2020 -0300

    Create user endpoints and user sessions

commit 07fad6135e58d55174cfb183d9837614298b6470
Author: Alain Dupuis <adupuis2@unb.ca>
Date:   Sun Mar 8 17:51:49 2020 -0300

    Create user endpoints and user sessions
```

jasonncleveland commented on src/rickbox/user_content/USERID_1_cookie-jar:
```diff
@@ -0,0 +1,5 @@
+# Netscape HTTP Cookie File
```
Does this file need to be committed?

adupuis2 replied:
Nope, slipped in. Whoops!

jasonncleveland commented on src/rickbox/resources/file_manager.py:
```diff
@@ -0,0 +1,62 @@
+            print(request)
+            return ERROR_400
+
+        # SQL injection?
```
The save function calls the stored procedure with parameterized input.  Should be no SQL injection risk.

adupuis2 committed:
Remove SQL Injection comment
Remove user_content files that slipped in
